python/ray/serve/_private/deployment_state.py

```diff
@@ -1204,7 +1206,7 @@ def recover_current_state_from_replica_actor_names(
                 self._detached,
                 replica_name.replica_tag,
                 replica_name.deployment_tag,
-                None,
+                self._target_state.version,
```

Comment:
can you help me to understand how do we guarantee version is not None?

Reply:
Previously, version was not used in constructing a `DeploymentReplica` instance, and is instead populated in `check_ready()` when `ready_obj_ref` returns with metadata including the version. 
Now, version is always passed in when constructing a `DeploymentReplica` instance. This will be an accurate version when we are starting a new replica (i.e. `replica = DeploymentReplica(version), replica.start()`), but it may be inaccurate when we are recovering a replica (i.e. `replica = DeploymentReplica(version), replica.recover()`). However, we can guarantee that version will never be None in a DeploymentReplica instance.

Also, the target state of a deployment should never have a null version (unless the target state hasn't been set yet, in which case nothing is starting replicas).